library/gensym.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule 
+
+def test_name(name, used_list):
+    for item in used_list:
+        if name in item:
+            return False
+    return True
+
+def init_volume(result, module):
+    # Check if fs_type is 'swap', use that instead of mount pt
+    if 'swap' in module.params['fs_type'].lower():
+        result['volume_name'] = 'swap'
+
+    # Check if the mount point of fs is root 
+    else:
+        if module.params['mount'] == '/':
+            result['volume_name'] = 'root'
+        else:
+            result['volume_name'] = module.params['mount'][1:].replace('/', '_')
+
+    def get_lvcmd_buf():
+        lvs_cmd = subprocess.check_output('sudo lvs --no-headings -o name', shell = True)
```

Comment:
It would be better to use the facts for this. I think you can do something along the lines of:

```python
from module_utils import facts
...
lvm_facts = facts.ansible_facts(module, gather_subset=set(['lvm']))
vg_names = lvm_facts['vgs'].keys()
lv_names = lvm_facts['lvs'].keys()
```
I haven't actually tried it, but something like that should work.

The same goes for the code below that gets the list of vg names.